hazelcast/src/test/java/com/hazelcast/config/ExternalDataStoreConfigTest.java

```diff
@@ -37,4 +48,48 @@ public void testEqualsAndHashCode() {
                 .suppress(Warning.NONFINAL_FIELDS)
                 .verify();
     }
+
+    @Test
+    public void should_serialize_with_empty_properties() throws IOException {
+        ExternalDataStoreConfig originalConfig = new ExternalDataStoreConfig()
+                .setName("some-name")
+                .setClassName("some-class-name")
+                .setShared(false);
+
+        byte[] data = writeData(originalConfig);
+        ExternalDataStoreConfig deserializedCopy = readData(data);
+
+        assertThat(deserializedCopy).isEqualTo(originalConfig);
+    }
+
+    @Test
+    public void should_serialize_with_NON_empty_properties() throws IOException {
+
+        Properties properties = new Properties();
+        properties.setProperty("prop1", "val1");
+        properties.setProperty("prop2", "val2");
+        ExternalDataStoreConfig originalConfig = new ExternalDataStoreConfig()
+                .setName("some-name")
+                .setClassName("some-class-name")
+                .setProperties(properties);
+
+        byte[] data = writeData(originalConfig);
+        ExternalDataStoreConfig deserializedCopy = readData(data);
+
+        assertThat(deserializedCopy).isEqualTo(originalConfig);
+    }
+
+    private ExternalDataStoreConfig readData(byte[] data) throws IOException {
+        BufferObjectDataInput objectDataInput = serializationService.createObjectDataInput(data);
+        ExternalDataStoreConfig deserializedCopy = new ExternalDataStoreConfig();
+        deserializedCopy.readData(objectDataInput);
+        return deserializedCopy;
+    }
+
+    private byte[] writeData(ExternalDataStoreConfig originalConfig) throws IOException {
+        BufferObjectDataOutput objectDataOutput = serializationService.createObjectDataOutput();
+        originalConfig.writeData(objectDataOutput);
+        byte[] data = objectDataOutput.toByteArray();
+        return data;
+    }
```

Comment:
Why not use serialization service's `toData` and `toObject` methods instead of these?

Reply:
fixed: https://github.com/hazelcast/hazelcast/pull/21930/commits/ca886aee6fa19ee7af2bad48af7320ca6437b90f